code/game/machinery/washing_machine.dm

```diff
@@ -39,8 +39,27 @@
 
 	busy = TRUE
 	update_icon()
+	addtimer(CALLBACK(src, .proc/wiggle))
 	addtimer(CALLBACK(src, .proc/wash_cycle), 200)
 
+/obj/machinery/washing_machine/proc/wiggle()
+	if (!busy)
+		animate(src, transform=matrix(), time=2)
+		return
+	if (anchored)
+		if (prob(5))
+			var/matrix/M = new
+			M.Translate(rand(-1, 1), rand(0, 1))
+			animate(src, transform=M, time=1)
+			animate(transform=matrix(), time=1)
+	else
+		if (prob(1))
+			step(src, pick(GLOB.cardinals))
+		var/matrix/M = new
+		M.Translate(rand(-3, 3), rand(-1, 3))
+		animate(src, transform=M, time=2)
+	addtimer(CALLBACK(src, .proc/wiggle), 2)

```

Comment:
Use a cancelable looping timer for this sort of thing

Reply:
How would that compare to using `SSfastprocess`, which has the appropriate `wait` of 2?

Reply:
If you have a subsystem you can use that would be better